File: ClientePy/control/ControlParseJSON.py
import datetime
import json

from control import CustomEncoder


class ControlParseJSON:

    def parseJSON(self, obj):
        return json.dumps(obj, indent=4, cls=CustomEncoder)

    def fromJSON(self, jsonSTR, obj):
        return json.loads(jsonSTR, object_hook=self.decode_object)

    def decode_object(self, o):
        if '__A__' in o:
            a = o.A()

            a.__dict__.update(o['__A__'])

            return a

        elif '__datetime__' in o:
            return datetime.strptime(o['__datetime__'], '%Y-%m-%dT%H:%M:%S')


        return o

File: Servidor/src/br/control/ControlParseJSON.java
package control;

import com.google.gson.Gson;
import entity.Request;

public class ControlParseJSON {

    private Gson gson;

    public ControlParseJSON(){
        this.gson = new Gson();
    }

    public String parseJSON(Object t){
        return this.gson.toJson(t);
    }

    public Object fromJSON(String json, Class c){
        return gson.fromJson(json, c);
    }
}